lib/src/main/scala/spinal/lib/Stream.scala

```diff
@@ -549,9 +581,29 @@ class StreamFork[T <: Data](dataType: HardType[T], portCount: Int) extends Compo
   }
 }
 
+/**
+ * A simpler version of StreamFork. All output streams will always fire together,
+ *  which means that the stream will halt until all output streams are ready. In contrast,
+ *  the normal StreamFork allows that output streams may be ready one at a time, at the cost
+ *  of an additional flip flop (1 bit per output).
+ */
+// TODO this can be integrated into the normal StreamFork by adding a boolean parameter. But I don't fully understand StreamFork. ~piegames
```

Comment:
When there is function aguement conflicts in scala Object, i generaly just switch "apply" to a more expressive name, in the given case, could be  :

```scala
object StreamFork{
...
   def vec[T <: Data](source: Stream[T], count: Int): Vec[Stream[T]] 
} 

```

Reply:
About StreamFork, it take a stream, and each transaction is replicated on each output streams